src/md.js

```diff
@@ -1,23 +1,46 @@
 import template from "./template";
 import marked from "marked";
 
+const HL_ROOT =
+  "https://cdn.jsdelivr.net/npm/@observablehq/highlight.js@2.0.0/";
+
 export default function(require) {
   return function() {
-    return template(function(string) {
-      var root = document.createElement("div");
-      root.innerHTML = marked(string, {langPrefix: ""}).trim();
-      var code = root.querySelectorAll("pre code[class]");
-      if (code.length > 0) {
-        require("@observablehq/highlight.js@1.1.1/highlight.min.js").then(function(hl) {
-          code.forEach(function (block) {
-            hl.highlightBlock(block);
-            block.parentNode.classList.add('observablehq--md-pre');
+    return template(
+      function(string) {
+        var root = document.createElement("div");
+        root.innerHTML = marked(string, { langPrefix: "" }).trim();
+        var code = root.querySelectorAll("pre code[class]");
+        if (code.length > 0) {
+          require(HL_ROOT + "highlight.min.js").then(function(hl) {
+            code.forEach(function(block) {
+              function done() {
+                hl.highlightBlock(block);
+                block.parentNode.classList.add("observablehq--md-pre");
+              }
+              if (hl.getLanguage(block.className)) {
+                done();
+              } else {
+                require(HL_ROOT + "async-languages/index.js")
+                  .then(index => {
+                    if (index.has(block.className)) {
+                      return require(HL_ROOT +
+                        "async-languages/" +
+                        index.get(block.className)).then(language => {
+                        hl.registerLanguage(block.className, language);
+                      });
+                    }
+                  })
+                  .then(done, done);
```

Comment:
Actually, what would the async/await version of this look like? Realized it's ~3 nested promises.

Reply:
It'd be a little shinier but require a change in the support expectations of this module - it's currently in a very conservative style, even using old-fashioned `var`. That's what I had initially but decided that this wasn't enough of a convenience to change the expectations of the module, and if we do, we'll want to use all the newfangled other stuff too.

Reply:
ah right, forgot which module I was looking at.